src/api/test-controller/index.js

```diff
@@ -389,15 +389,24 @@ export default class TestController {
         return this.testRun.executeAction(name, new GetBrowserConsoleMessagesCommand(), callsite);
     }
 
-    _checkForExcessiveAwaits (selectorCallsiteList, expectCallsite) {
-        for (const selectorCallsite of selectorCallsiteList) {
-            if (selectorCallsite.filename === expectCallsite.filename &&
-                selectorCallsite.lineNum === expectCallsite.lineNum) {
-                addWarning(this.warningLog, WARNING_MESSAGE.excessiveAwaitInAssertion, selectorCallsite);
+    // NOTE: Gets the callsite stackFrame string representation: 'filename:lineNum:colNum'
+    _getCallsiteStackFrameString (callsite) {
+        return callsite.stackFrames[callsite.callsiteFrameIdx].toString();
+    }
 
-                selectorCallsiteList.delete(selectorCallsite);
-            }
+    _checkForExcessiveAwaits (snapshotPropertyCallsites, callsiteToCheck) {
+        const key = `${callsiteToCheck.filename}:${callsiteToCheck.lineNum}`;
```

Comment:
`const callsiteId = ...`

Reply:
I'm not sure if we should call it an ID since there can be callsites with the matching filename:lineNum pair

Reply:
I agree with you about ID. If you can imagine a better name - you are welcome.

Reply:
Would `callsiteLineId` (or just `lineId`) be better?